### PR TITLE
Fixed Login and Implemented Rate Limit hotfix

### PR DESCRIPTION
**Pull request by SippieCup**

### CODE REVIEW

**Comment by SippieCup** [comment]
The Old PR https://github.com/SwipeX/PokeMate/pull/194 I screwed up because I havent slept in 42 hours and accidently merged a bunch of my other stuff into it.

Jitpack should be working now so compiling from source should be a breeze once the snapshot is updated to the 300ms patch.

**Comment by pharrowboy** [comment]
the delay added in by the 300 ms has made it so that the app is limited to a moving speed of 0.5. i had mine set to 4, my character walked all around and didn't catch anything after walking for 10 minutes. changed it to 0.5 it walked for 10 seconds then it found a drowsie.

**Comment by SwipeX** [comment]
Can you explain to me what exactly this does??

File: src/main/java/dekk/pw/pokemate/Context.java [hunk 1]
```diff
@@ -66,8 +67,16 @@ public static CredentialProvider Login(Context context, OkHttpClient httpClient)
...
                     token = scanner.nextLine();
                     scanner.close();
-                    if (token != null) {
-                        return new GoogleAutoCredentialProvider(httpClient, Config.getUsername(), Config.getPassword());
+                    if (token != null && Config.getUsername().contains("@")) {
+                            GoogleUserCredentialProvider provider = new GoogleUserCredentialProvider(httpClient);
+                            System.out.println("-----------------------------------------");
+                            System.out.println("  Please go to the following URL");
+                            System.out.println(GoogleUserCredentialProvider.LOGIN_URL);
+                            System.out.println("Enter authorisation code:");
```
Comment (SwipeX) [inline src/main/java/dekk/pw/pokemate/Context.java:75]: authorization (sp)

**Pull request merged**